Project/ClientApp/src/components/Login.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import { useUserContext } from "../userContext";
+
+const Login = () => {
+  const { userEmail, setUserEmail, username, setUsername } = useUserContext();
+  const [loginInputEmail, setLoginInputEmail] = useState("");
+  const [loginInputPassword, setLoginInputPassword] = useState("");
+  const [registerInputEmail, setRegisterInputEmail] = useState("");
+  const [registerInputName, setRegisterInputName] = useState("");
+  const [registerInputSurname, setRegisterInputSurname] = useState("");
+  const [registerInputPassword, setRegisterInputPassword] = useState("");
+
+  const requestBody = {
+    userEmail: loginInputEmail,
+    userPassword: loginInputPassword,
+  };
+
+  const handleLogin = async () => {
+    if (!loginInputEmail || !loginInputPassword) {
+      return;
+    } else {
+      const response = await fetch("https://localhost:7015/user/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(requestBody),
+      });
+  
+      if (!response.ok) {
+        alert("Incorrect email or password");
+        return;
+      }
+  
+      const loginToken = await response.text();
+      localStorage.setItem("loginToken", loginToken);
+      const userInfo = await fetch("https://localhost:7015/user/get/" + loginToken, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      const userData = await userInfo.json();
+      console.log(userData.name);
+
+      window.location.href = "/";
+    }
+  };
+
+  const validateEmail = (email) => {
+    return String(email)
+      .toLowerCase()
+      .match(
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+      );
+  };
```

Comment:
Regex validation should be done in C# (can make helper folder that will have helper classes. Example: RegexValidation.cs that has function with regex to validate for example an email)

Reply:
Won't that not allow to print an error message to the user if the email is invalid (for example a javascript alert)?

Reply:
You are able to create your own OK or BadRequest with data inside of it, for example if the request was successful or not and a message
`return Ok(new {success = true, message = "Logged in"`
or
`return BadRequest(new {success = false, message = "Invalid email or password"`

Reply:
Ok, I'll get onto that later